From: Jan Engelhardt <jengelh@inai.de>
Date: 2013-03-17 14:43:46.875421301 +0100

build: resolve order of link arguments

Libraries do not belong into flags!
User variables should not be overriden!
Resolve the mess by using pkg-config to determine SDL's location.
---
 configure.ac    |    4 +---
 src/Makefile.am |    2 ++
 2 files changed, 3 insertions(+), 3 deletions(-)

Index: tcd-2.2.0/configure.ac
===================================================================
--- tcd-2.2.0.orig/configure.ac
+++ tcd-2.2.0/configure.ac
@@ -13,9 +13,7 @@ AC_ARG_ENABLE([debug],
 AC_PROG_CC
 
 # Checks for libraries
-ri_CONFIG_SCRIPT(sdl, 1.2.0, SDL_CPPFLAGS, SDL_LDFLAGS)
-CPPFLAGS="$CPPFLAGS $SDL_CPPFLAGS"
-LDFLAGS="$LDFLAGS $SDL_LDFLAGS"
+PKG_CHECK_MODULES([SDL], [sdl >= 1.2])
 
 AC_CHECK_LIB(ncurses, initscr, , AC_MSG_ERROR([No ncurses library found.]))
 
Index: tcd-2.2.0/src/Makefile.am
===================================================================
--- tcd-2.2.0.orig/src/Makefile.am
+++ tcd-2.2.0/src/Makefile.am
@@ -1,4 +1,6 @@
+AM_CPPFLAGS = ${SDL_CFLAGS}
 bin_PROGRAMS = tcd
 man_MANS = tcd.1
 
 tcd_SOURCES = cd-utils.c cd-utils.h cddb.c concat-strings.c tcd.c tracked.c user-interface.c cddb.h concat-strings.h tcd.h tracked.h user-interface.h ui-layout.c ui-layout.h
+tcd_LDADD = ${SDL_LIBS}
